File: GGAndroid_2.0/GGDemo/src/main/java/io/ganguo/demo/viewmodel/item/ItemQiNiuImageVModel.java
package io.ganguo.demo.viewmodel.item;

import android.databinding.ObservableField;
import android.view.View;

import io.ganguo.demo.R;
import io.ganguo.demo.databinding.ItemQiniuImageBinding;
import io.ganguo.vmodel.BaseViewModel;
import io.ganguo.library.ui.view.ViewInterface;

/**
 * <p>
 * 七牛demo - item ViewModel
 * </p>
 * Created by leo on 2018/7/30.
 */
public class ItemQiNiuImageVModel extends BaseViewModel<ViewInterface<ItemQiniuImageBinding>> {
    public ObservableField<String> imageUrl = new ObservableField<>();
    public ObservableField<String> index = new ObservableField<>();

    public ItemQiNiuImageVModel(String imageUrl, String index) {
        this.imageUrl.set(imageUrl);
        this.index.set(index);
    }

    @Override
    public int getItemLayoutId() {
        return R.layout.item_qiniu_image;
    }

    @Override
    public void onViewAttached(View view) {

    }
}

File: GGAndroid_3.0/main/app-mvvm-v1-sample/src/main/java/io/ganguo/demo/viewmodel/item/ItemQiNiuImageVModel.kt
package io.ganguo.demo.viewmodel.item

import androidx.databinding.ObservableField
import android.view.View

import io.ganguo.demo.R
import io.ganguo.demo.databinding.ItemQiniuImageBinding
import io.ganguo.viewmodel.core.BaseViewModel
import io.ganguo.viewmodel.core.viewinterface.ViewInterface

/**
 *
 *
 * 七牛demo - item ViewModel
 *
 * Created by leo on 2018/7/30.
 */
class ItemQiNiuImageVModel(imageUrl: String, index: String) : BaseViewModel<ViewInterface<ItemQiniuImageBinding>>() {
    var imageUrl = ObservableField<String>()
    var index = ObservableField<String>()

    init {
        this.imageUrl.set(imageUrl)
        this.index.set(index)
    }


    override fun onViewAttached(view: View) {

    }

    override val layoutId: Int by lazy {
        R.layout.item_qiniu_image
    }
}
